### PR TITLE
Remove kickstart sources from the class DNFPayload (2/5)

### PR DESCRIPTION
**Pull request by poncovka**
This is the second part of the code that will replace RPM sources in UI with
the DBus sources of the Payloads module. All parts have to be tested
and merged together.

Remove kickstart sources from the class DNFPayload.  Use the DBus sources
instead.

**DO NOT MERGE!**

### CODE REVIEW

File: pyanaconda/payload/dnf/payload.py [hunk 1]
```diff
@@ -139,9 +140,19 @@ def proxy(self):
...
+        """
+        return get_source(self.proxy, SOURCE_TYPE_REPO_FILES)
+
     @property
     def is_hmc_enabled(self):
-        return self.data.method.method == "hmc"
+        # FIXME: Remove this property and check the type directly.
+        source_proxy = self.get_source_proxy()
+        source_type = source_proxy.Type
+        return source_type == SOURCE_TYPE_HMC
```
Comment (jkonecny12) [inline pyanaconda/payload/dnf/payload.py:155]: I would merge these two lines. I don't see reason here to have it on two lines.

Comment (poncovka) [reply pyanaconda/payload/dnf/payload.py]: Fixed.

**Comment by poncovka** [comment]
Rebased.

File: pyanaconda/payload/dnf/payload.py [hunk 2]
```diff
@@ -563,6 +557,23 @@ def _configure_proxy(self):
             config.proxy_username = None
             config.proxy_password = None
 
+    def _get_proxy_url(self):
```
Comment (jkonecny12) [inline pyanaconda/payload/dnf/payload.py:601]: Please rename this to `_get_url_source_proxy`.

Comment (poncovka) [reply pyanaconda/payload/dnf/payload.py:604]: It is not a DBus proxy, but the URL proxy.

**Comment by poncovka** [comment]
Fixed some issues.

**Comment by poncovka** [comment]
Fixed the error handling.

**Comment by poncovka** [comment]
Small fixes in logging.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Added fix for metalink type @poncovka if you want you can squash it.

**Comment by poncovka** [comment]
@jkonecny12, that is fine. Thanks for the fix!

**Pull request merged**